src/main/java/com/greenfoxacademy/domain/GenericToken.java

```diff
@@ -23,52 +23,42 @@
     @Temporal(value = TemporalType.TIMESTAMP)
     Date valid;
 
+
     @Transient
-    final int DEFAULT_TIMEOUT = 30;
+    final int DEFAULT_TIMEOUT = 30*60*1000;
 
     public GenericToken() {
-        this.timestamp = new Date();
-        Calendar cal = Calendar.getInstance();
-        cal.setTime(this.timestamp);
-        cal.add(Calendar.MINUTE, DEFAULT_TIMEOUT);
-        this.valid = cal.getTime();
+        long current_time=System.currentTimeMillis();
+        this.timestamp=new Date(current_time);
+        this.valid=new Date(current_time+DEFAULT_TIMEOUT);
     }
 
-
     public GenericToken(int minutes) {
-        this.timestamp = new Date();
-        Calendar cal = Calendar.getInstance();
-        cal.setTime(this.timestamp);
-        cal.add(Calendar.MINUTE, minutes);
-        this.valid = cal.getTime();
+        long current_time=System.currentTimeMillis();
+        this.timestamp=new Date(current_time);
+        this.valid=new Date(current_time+minutes);
```

Comment:
@viktornagy for example here. minute is not the format of the int what comes in, its millisec, do the multiplying here instead of where you call this.